### ISSUE TITLE
Trigger notifications for changes on project plan

### ISSUE THREAD

**Issue by vic-topcoder** [appirio-tech/connect-app#2178]
### Expected behavior
Activities and changes of project plan should trigger notifications to drive users' attention and guide them back to the UI.

- Project plan is ready (when we first publish a phase in Planned mode from draft)
- Project phase changes
  - Phase status change to Active (your phase is now active)
  - Phase status change to Delivered (we've closed the phase)
  - Phase progress changed
  - Phase payment changed
  - Phase new posts (should work as regular posts, and load the phase expanded, with tab selected and latest messages visible)
  - Scope changes on a phase - updated, files added
- Project progress changes

**Comment by vikasrohit** [comment]
@vic-topcoder @rmjmccormick  @gondzo  I think we should also handle the email notifications for these new events at the same time. https://github.com/topcoder-platform/tc-notifications/issues/51

**Comment by vic-topcoder** [comment]
Agreed - anytime I mention Notification, we should trigger all channels (in-app, email, and future {push, desktop, browser tab counter, etc.})

**Comment by architectt1** [comment]
@gondzo @vic-topcoder

> Project plan is ready (when we first publish a phase in Planned mode from draft)

Should this be triggered for both `create` and `update` actions for a given phase? I'm not sure if *publish* refers to `create` or to `update` as well. Also, I'm assuming that, if a phase already exists with the Planned mode for the project in question, then we shouldn't send this notification. Is that correct?

> Scope changes on a phase - updated, files added

Regarding the files added, there's a `project_attachments` table in `tc-project-service`, is that what is being mentioned in here? Note that they associate to a project, not to a particular phase. What is the wanted approach for this requirement?

Let me know if this is not the correct place for this. These are changes related to https://github.com/topcoder-platform/tc-project-service, but the issue was present in here.

**Comment by gfzabarino** [comment]
@gondzo @vic-topcoder I also noticed that in the project's API, an authorized user can create phases in any state. For all the phase transitions' notifications, these should be about `update` and not `create` actions, right?

**Comment by architectt1** [comment]
@gondzo @vic-topcoder I also noticed that in the project's API, an authorized user can create phases in any state. For all the phase transitions' notifications, these should be about update and not create actions, right?

**Comment by architectt1** [comment]
Could you please add the following topics to Kafka (@gondzo that would be the instance you passed me access credentials about)?

```
notifications.connect.project.planModified (this is already in dev, but it seems is not registered)
notifications.connect.project.planReady
project.phase.transition.active
project.phase.transition.completed
project.phase.update.payment
project.phase.update.progress
project.phase.update.scope
notifications.connect.project.progressModified
```

For files, we already have the `notifications.connect.project.fileUploaded` event being sent, but again, that's related to the project as a whole, and not to a particular phase. @gondzo if you can, please double check this event is already added to Kafka or not, as I didn't yet test it.

**Comment by architectt1** [comment]
@gondzo @vic-topcoder I was able to install a local Kafka instance, so I'm able to test locally. I need the following information from you, as I don't have a deep insight of the business logic of the system: for each Kafka event, please let me know which roles from the ones below need to get notified:

### projectRoles

* `owner`
* `copilot`
* `manager`
* `member`

### topcoderRoles

* `Connect Copilot`
* `Connect Manager`
* `administrator`

### Kafka events

```
notifications.connect.project.planReady
project.phase.transition.active
project.phase.transition.completed
project.phase.update.payment
project.phase.update.progress
project.phase.update.scope
notifications.connect.project.progressModified
```

**Comment by architectt1** [comment]
I can confirm attachments added to projects are generating a notification record to all project's members with notification type `notifications.connect.project.fileUploaded`.

**Comment by vikasrohit** [comment]
@gondzo Please keep the changes for this issue in a separate feature branch because it is not scheduled for upcoming release.

> Project plan is ready (when we first publish a phase in Planned mode from draft)

@vic-topcoder Should we say that Project Plan is ready when only one phase in the plan is set to planned mode? May be we should send this event when all phases in the project plan are marked as planned?
Also, if we raise both `Phase progress changed` and `Project progress changes`, it would trigger two events when a phase's progress is changes because whenever progress of phase changes, it changes the progress of the project as well. In fact we don't have any other event which can change the progress of the project.

**Comment by gondzo** [comment]
We have the PR ready in project service https://github.com/topcoder-platform/tc-project-service/pull/110/files

**Comment by vikasrohit** [comment]
Thanks @gondzo. I have merged the PR, I think it has safe changes, it would only generate web notifications for now and for email notifications we already handling via https://github.com/topcoder-platform/tc-notifications/issues/59.

**Comment by vikasrohit** [comment]
@gondzo Could you please get the changes required in `connect-app` to show the web notifications for all these new events? I think we might need to have a way to navigate to the specific phase from the notifications i.e. a new way to goto specific area of the application [here](https://github.com/appirio-tech/connect-app/blob/781f0877599ae22a86837be10515de15f86b300c/src/routes/notifications/constants/notifications.js#L13).

**Comment by vikasrohit** [comment]
Moving it to next release as it does not seem that we can complete this today.

**Comment by RishiRajSahu** [comment]
@maxceem All the web notifications for Project-Plan are getting generated correctly, Now we need two tasks here
1.Making them rendered on UI as notifications
2.Adding Navigation capability for Project-Plan notifications

**Comment by maxceem** [comment]
@vikasrohit @RishiRajSahu 

In [this sheet](https://docs.google.com/spreadsheets/d/15u4cwCg5BwgyJkVsd-wp_T2pUnDrqW2bEZuwAn3qArA/edit#gid=0) we don't have texts for the next notifications:
- Phase progress changed (`project.phase.update.progress`)
- Phase payment changed (`project.phase.update.payment`) 

Also, in the sheet we have text for the next notifications:
- Milestone is completed
- Milestone is active
- Milestone awaits customer input
But don't have such events triggered in `tc-project-service`. Should we add them in project service or wait with it?

**Comment by vikasrohit** [comment]
@maxceem we are working on adding more texts for new events.

Also, milestone events are now working in dev env. We did that last week only.

Just fyi, me and @gondzo are working on getting these new events and their notifications in connect. It is separate scope of work than what we are planning for https://github.com/appirio-tech/connect-app/issues/2344, though these both have an overlapping of the way events are raised.

**Comment by vikasrohit** [comment]
@gondzo I have added the configuration for showing the notifications for these new events in `connect-app` along with changes for #2468 in branch `feature/plan-timeline-events-settings`.

**Comment by vikasrohit** [comment]
@vic-topcoder  @acshields should we consider updates in the Price field of phase for `Phase payment changed` notification? or would it be fine just to capture the changes in `Paid To Date` field for this notification?
